src/neighbors/kNeighborsRegressor.test.ts

```diff
@@ -13,65 +13,17 @@
 * ==========================================================================
 */
 
-import { dfd, tf } from '../shared/globals'
-import { meanSquaredError } from '../metrics/metrics'
-import { Tensor1D, Tensor2D } from '@tensorflow/tfjs-core'
 import { KNeighborsRegressor } from './kNeighborsRegressor'
 import { KNeighborsParams } from './kNeighborsBase'
 import { fetchCaliforniaHousing, loadDiabetes } from '../datasets/datasets'
 import { arrayEqual } from '../utils'
-
-// TODO: replace this with KFold as soon as its implemented
-function* kFoldIndices(
-  k: number,
-  len: number
-): IterableIterator<[Int32Array, Int32Array]> {
-  if (!(0 < k && k % 1 === 0)) {
-    throw new Error('kFoldIndices(k,len): k must be a positive integer.')
-  }
-  k = Math.floor(k)
-
-  if (!(k <= len && len % 1 === 0)) {
-    throw new Error('kFoldIndices(k,len): len must be an int greater than k.')
-  }
-  len = Math.floor(len)
-
-  const n = Math.floor(len / k)
-  let rem = len % k
-
-  const range = new Int32Array(len)
-  for (let i = len; i-- > 0; ) {
-    range[i] = i
-  }
-
-  let off = 0
-  for (let chunk: number; off < len; off += chunk) {
-    chunk = n + Number(rem-- > 0)
-
-    const test = new Int32Array(len - chunk)
-    test.set(range.subarray(0, off), 0)
-    test.set(range.subarray(off + chunk), off)
-
-    yield [test, range.slice(off, off + chunk)]
-  }
-
-  expect(off).toEqual(len)
-}
-
-function* kFold(
-  k: number,
-  X: Tensor2D,
-  y: Tensor1D
-): IterableIterator<[Tensor2D, Tensor1D, Tensor2D, Tensor1D]> {
-  const len = X.shape[0]
-  if (len !== y.shape[0]) {
-    throw new Error('kFold(k,X,y): X.shape[0] must equal y.shape[0].')
-  }
-
-  for (const [train, test] of kFoldIndices(k, len)) {
-    yield [X.gather(train), y.gather(train), X.gather(test), y.gather(test)]
-  }
-}
+import { crossValScore } from '../model_selection/crossValScore'
+import { KFold } from '../model_selection/kFold'
```

Comment:
I love how squeaky clean this file got 👍 Good work on this whole PR @DirkToewe

Reply:
That was the motivation for taking on this PR :smile: